name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/TimelineChart.java

```diff
@@ -297,33 +302,42 @@ else if (days < 1200)
             // Sparse labeling mode used in low width conditions.
             // Its purpose is to ensure enough space between the labels to avoid overlays and ensure readability. 
             // It is mainly relevant to charts embedded into dashboards with narrow columns.
-            if (days < 120)
+            if (days < 100)
             {
                 period = Period.ofMonths(1);
                 format = DateTimeFormatter.ofPattern("MMMM yyyy"); //$NON-NLS-1$
             }   
-            else if (days < 250)
+            else if (days < 400)
             {
                 period = Period.ofMonths(3);
                 format = DateTimeFormatter.ofPattern("QQQ yyyy"); //$NON-NLS-1$
-                cursor = cursor.plusMonths((12 - cursor.getMonthValue() + 1) % 3);
```

Comment:
Why do you delete this line?

The "cursor" needs to be moved to the first occurrence of the time grid.
Let's say it is February and you want to paint quarterly lines, this needs to move to April (as the beginning of the second quarter).

I re-added the lines - otherwise the change looks fine and I will merge.

Reply:
Yes, you are right of course. Without the cursor alignment the quarter lines are not placed correctly.

![pp_quarter_line_regression](https://user-images.githubusercontent.com/23013422/148681068-322ed831-2b20-4a52-812a-868a922b1dc3.png)